src/v2/api/services/GroupService.ts

```diff
@@ -284,4 +285,63 @@ export class GroupService {
     }
   }
 
+  async switchCaptain (groupId: string, studentId: string) {
+    const isStudentBelongToGroup = await this.checkIsStudentBelongToGroup(groupId, studentId);
+    if (!isStudentBelongToGroup) throw new NoPermissionException();
+    const { id: oldCaptainId } = await this.getCaptain(groupId);
+
+    const newCaptainRole = await this.roleRepository.find({
+      groupRole: {
+        groupId,
+      },
+      userRoles: {
+        some: {
+          studentId,
+        },
+      },
+    });
+
+    const oldCaptainRole = await this.roleRepository.find({
+      groupRole: {
+        groupId,
+      },
+      userRoles: {
+        some: {
+          studentId: oldCaptainId,
+        },
+      },
+    });
+
+    await this.roleRepository.updateById(newCaptainRole.id, {
+      userRoles: {
+        updateMany: {
+          where: {
+            studentId,
+          },
+          data: {
+            studentId: oldCaptainId,
+          },
+        },
+      },
+    });
+
+    await this.roleRepository.updateById(oldCaptainRole.id, {
+      userRoles: {
+        updateMany: {
+          where: {
+            studentId: oldCaptainId,
+          },
+          data: {
+            studentId,
+          },
+        },
+      },
+    });
+  }
+
+  private async checkIsStudentBelongToGroup (groupId: string, studentId: string) {
```

Comment:
This method has long name, thus, it is better to rename `checkIsStudentBelongToGroup` to `isStudentInGroup`